Training/Training/TestReversal.c

```diff
@@ -0,0 +1,123 @@
+// ------------------------------------------------------------------------------------------------
+// Assignment4 ~ A training program for new joiners at Metamation, Batch - Sept 2024.
+// Copyright (c) Gokul TS @Metamation India.
+// ------------------------------------------------------------------
+// TestReversal.c
+// C program to check test cases
+// ------------------------------------------------------------------------------------------------
+#include "Reversal.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+/// <summary>Gets string input from user</summary>
+void GetStringInput ();
+/// <summary>Gets integer input from user</summary>
+void GetIntegerInput ();
+/// <summary>Checks test cases</summary>
+void OutputCheck ();
+/// <summary>Clears the content on screen</summary>
+void ClrScreen ();
+
+int main () {
+   while (1) {
+      char option[3];
+      printf ("1-PALINDROME CHECKER\n2-REVERSE A NUMBER\n3-TEST CASES\n4-CLEAR SCREEN\n5-EXIT\n");
+      printf ("Enter an option:");
+      fgets (option, sizeof (option), stdin);
+      char* Result = strchr (option, '\n');
+      if (Result == NULL) {
+         while (getchar () != '\n');
+         printf (ANSI_COLOR_RED"Please enter a number between 1-5\n\n"ANSI_RESET_ALL);
+      } else {
+         int num = atoi (option);
+         switch (num) {
+            case 1:
+               GetStringInput ();
+               break;
+            case 2:
+               GetIntegerInput ();
+               break;
+            case 3:
+               OutputCheck ();
+               break;
+            case 4:
+               ClrScreen ();
+               break;
+            case 5:
+               return 0;
+         }
+      }
+   }
+   return 0;
+}
+
+void GetStringInput () {
+   char phrase[MAX_CHAR];
+   printf ("\nEnter a Phrase: ");
+   char* j = fgets (phrase, sizeof (phrase), stdin);
+   char* strResult = strchr (phrase, '\n'); //To check whether input exceeds limit
+   if (strResult == NULL) while (getchar () != '\n');
+   if (j != NULL && phrase[0] != '\n' && strResult != NULL) {

```

Comment:
```suggestion
   if (j == NULL || phrase[0] == '\n' || strResult == NULL) continue;

```
See [Guard](https://en.wikipedia.org/wiki/Guard_(computer_science)).